src/Test/L0/Worker/PluginInternalUpdateRepositoryPathCommandL0.cs

```diff
@@ -0,0 +1,203 @@
+                Assert.Equal(null, _variables.Get(Constants.Variables.System.DefaultWorkingDirectory));
+            }
+        }
+
+        private TestHostContext CreateTestContext([CallerMemberName] String testName = "")
+        {
+            var hc = new TestHostContext(this, testName);
+            _expressionManager = new ExpressionManager();
+            _expressionManager.Initialize(hc);
+            return hc;
+        }
+
+        private void InitializeExecutionContext(TestHostContext hc, bool isMultiCheckout = false)
+        {
+            List<string> warnings;
+            _variables = new Variables(
+                hostContext: hc,
+                copy: new Dictionary<string, VariableValue>(),
+                warnings: out warnings);
+            _repositories = new List<RepositoryResource>();
+            _selfRepo = new RepositoryResource()
+            {
+                Alias = "self",
+                Id = Guid.NewGuid().ToString(),
+                Name = "mainRepo",
+                Type = "git",
+            };
+            _selfRepo.Properties.Set(RepositoryPropertyNames.Path, "/1/s");
+            _otherRepo = new RepositoryResource()
+            {
+                Alias = "repo2",
+                Id = Guid.NewGuid().ToString(),
+                Name = "otherRepo",
+                Type = "git",
+            };
+            _otherRepo.Properties.Set(RepositoryPropertyNames.Path, "/1/otherRepo");
+            _repositories.Add(_selfRepo);
+            _repositories.Add(_otherRepo);
+
+            _ec = new Mock<IExecutionContext>();
+            _ec.SetupAllProperties();
+            _ec.Setup(x => x.Variables).Returns(_variables);
+            _ec.Setup(x => x.Repositories).Returns(_repositories);
+            _ec.Setup(x => x.GetHostContext()).Returns(hc);
+            _ec.Setup(x => x.SetVariable(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<bool>(), It.IsAny<bool>(), It.IsAny<bool>(), It.IsAny<bool>()))
+                .Callback<string, string, bool, bool, bool, bool>((name, value, secret, b2, b3, readOnly) => _variables.Set(name, value, secret, readOnly));
+
+            if (isMultiCheckout)
+            {
+                var jobSettings = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+                jobSettings.Add(Agent.Sdk.WellKnownJobSettings.HasMultipleCheckouts, Boolean.TrueString);
+                _ec.Setup(x => x.JobSettings).Returns(jobSettings);
+            }
+
+            var directoryManager = new Mock<Microsoft.VisualStudio.Services.Agent.Worker.Build.IBuildDirectoryManager>();
+            directoryManager.Setup(x => x.GetRelativeRepositoryPath(It.IsAny<string>(), It.IsAny<string>()))
+                .Returns<string, string>((bd, path) => GetLastPathPart(path));
+            
+            hc.SetSingleton(directoryManager.Object);
+        }
+
+        private string GetLastPathPart(string path)
```

Comment:
is this the same as ```Path.GetFileName(path)``` ?

Reply:
Unfortunately, Path looks at the OS to determine the separator. This function uses either separator so my hard coded strings work. I could probably change everything in the tests to use Path.Combine as well, but it gets really verbose.

Reply:
gotcha, I hate that about Path.